[KOGITO-8818] Rearrange guides to add operator and quarkus cloud (#313)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/cloud/operator/build-and-deploy-applications.adoc

> = Building and Deploying Production Ready Workflows with the Operator

Reviewer: Not sure about the term production ready?  Wondering if it gives people the impression things are supported?

Author: What term do you suggest? They must mirror. So if we have the "Development Profile", then we have "Production Profile". Mutable/Immutable?

> Building and Deploying Workflows in Production environments with the Operator

?

Author: For now, I've removed any "production" mentions and left only "build and deploy".

:compat-mode!:
// Metadata:

> :description: Build and deploy using the Kogito Serverless Workflow Operator a serverless workflow application

Reviewer 2: ```suggestion
:description: Build and deploy using Serverless Workflow Applications with Kogito Serverless Workflow Operator
```

> :keywords: kogito, workflow, serverless, operator, kubernetes, minikube

Reviewer 2: openshift and containers?

// links
:kogito_serverless_operator_url: https://github.com/kiegroup/kogito-serverless-operator/
:kogito_greeting_example_url: https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-greeting-quarkus
:kaniko_issue_url: https://github.com/GoogleContainerTools/kaniko/issues/2201

> This document describes how to build and deploy your workflow application using a Kubernetes cluster with the link:{kogito_serverless_operator_url}[{operator_name}]. 

Reviewer 2: ```suggestion
This document describes how to build and deploy your workflow on a Kubernetes cluster using the link:{kogito_serverless_operator_url}[{operator_name}]. 
```

> Using the {product_name} Operator, you will be able to build and deploy a workflow application only by having a workflow definition.

Reviewer 2: maybe delete and append the following statement in the previous paragraph?
```
only by having a workflow definition.
``

.Create a namespace for the application to build & run in
[source,bash,subs="attributes+"]
----
kubectl create namespace kogito-workflows
----

=== Create a secret for the container registry authentication
.Create a secret for the container registry authentication
[source,bash,subs="attributes+"]
----
kubectl create secret docker-registry regcred --docker-server=<registry_url> --docker-username=<registry_username> --docker-password=<registry_password> --docker-email=<registry_email> -n kogito-workflows
----

or you can directly import your local docker config into your Kubernetes cluster:

.Create a secret for the container registry authentication based on local docker config
[source,bash,subs="attributes+"]
----
kubectl create secret generic regcred --from-file=.dockerconfigjson=$\{HOME\}/.docker/config.json --type=kubernetes.io/dockerconfigjson -n kogito-workflows
----

[WARNING]
====
Double check your `$\{HOME\}/.docker/config.json`. If you're using local desktop authentication, this configuration **won't work** in the cluster. 
====

> === Create a {product_name} Platform containing the configuration (i.e. registry address, secret) for building your workflows

Reviewer: ```suggestion
=== Configure the {operator_name} (i.e. registry address, secret) for building your workflows
```

----
You can check the logs of the build of your workflow via:

.Get the workflow application pod logs
[source,bash,subs="attributes+"]
----
kubectl logs kogito-greeting-builder -n kogito-workflows
----

The final pushed image must be printed into the logs at the end of the build.

== Check the workflow application is running

In order to check that the {product_name} Greeting application is up and running, you can try to perform a test HTTP call, from the greeting pod.

.Check the greeting application is running
[source,bash,subs="attributes+"]
----
kubectl patch svc greeting -n kogito-workflows -p '{"spec": {"type": "NodePort"}}'
GREETING_SVC=$(minikube service greeting -n kogito-workflows --url)
curl -X POST -H 'Content-Type:application/json' -H 'Accept:application/json' -d '{"name": "John", "language": "English"}' $GREETING_SVC/greeting
----

If everything is working well you should receive a response like this:

.Response from the greeting application
[source,json,subs="attributes+"]
----
{"id":"b5fbfaa3-b125-4e6c-9311-fe5a3577efdd","workflowdata":{"name":"John","language":"English","greeting":"Hello from JSON Workflow, "}}
----

> // TODO: add a troubleshooting guide - https://issues.redhat.com/browse/KOGITO-8864

Reviewer 2: add the found a issue link?

Author: I don't follow. The link to the JIRA issue is there.
